### PR TITLE
Add a copy icon to entries in KeyValuesTable (#204)

### PR DESCRIPTION
**Pull request by everett980**
## Which problem is this PR solving?
- Tags are human-readable ( 🎉), but harder to paste into other tools.

## Short description of the changes
- Add icon to rows in KeyValueTable to copy that row's data.

204 copy tags with tooltip
(again, my converted gifs have a weird background color issue, looks fine in the browser and .webm recording)

Copied JSON:
```
{
  "key": "sql.query",
  "type": "string",
  "value": "SELECT * FROM customer WHERE customer_id=731"
}
```

I went with the copy icon over the word copy since I believe it is fairly ubiquitous, but open to changing.

### CODE REVIEW

**Comment by yurishkuro** [comment]
> I don't love that there is no "copy confirmation" so I welcome advice on how to indicate that to the user.

the copy icon can be animated to transform to a green checkbox.

Q: in what format is the tag being copied? I.e. how does the above example look after pasting it as text?

**Comment by everett980** [comment]
> Q: in what format is the tag being copied? I.e. how does the above example look after pasting it as text?

@yurishkuro the `JSON.stringify(row, null, 2)` looks like the following:
```
{
  "key": "hostname",
  "type": "string",
  "value": "6376c38cb700"
}
```
The `null, 2` arguments add new lines with two spaces for indentation. It helps considerably with legibility and just about any tool that accepts JSON input can handle the white-space characters.

**Comment by everett980** [comment]
> the copy icon can be animated to transform to a green checkbox.

> I would suggest using the tooltip on the icon to state / indicate:
> 
> * "Copy JSON" on mouseover
> * "Copied to clipboard" on click
> * * Set a mouseLeaveDelay to something like 500ms (or so) to let the user more easily read the message
> What do you think?

@yurishkuro @tiffon I went with the tooltip over the icon change as the tooltip will help with clarity for both before and after the copy event.
antd's tooltip struggles with a left-positioned tooltip's title increasing in length, so I used the title "Copy JSON" before the row has been copied, and "Copied" for after copy. I would hope that a user wouldn't be reliant on the "to clipboard" detail to know where copied data goes. I will update the gif in the description to include the tooltips.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.css [hunk 1]
```diff
@@ -40,6 +40,18 @@ limitations under the License.
...
 
+.KeyValueTable--copyColumn {
+  text-align: right;
+}
+
+.KeyValueTable--copyIcon {
+  visibility: hidden;
+}
+
+tr:hover > .KeyValueTable--copyColumn > .KeyValueTable--copyIcon {
```
Comment (tiffon) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.css]: Maybe a class should be added for the row? That will shorten the other selectors and avoid use of the element selector. Maybe the `KeyValueTable--body` class can be removed.

File: yarn.lock [hunk 2]
```diff
@@ -2701,6 +2702,13 @@ copy-template-dir@1.3.0:
     readdirp "^2.0.0"
     run-parallel "^1.1.4"
 
+copy-to-clipboard@^3:
+  version "3.0.8"
+  resolved "https://unpm.uberinternal.com/copy-to-clipboard/-/copy-to-clipboard-3.0.8.tgz#f4e82f4a8830dce4666b7eb8ded0c9bcc313aba9"
```
Comment (tiffon) [inline yarn.lock]: This needs to be hitting the yarn registry.

Comment (everett980) [reply yarn.lock]: updated

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.css (hunk 1 above)
Comment (everett980) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.css]: added `.KeyValueTable--row` and remove references to `tr` from .css file. still need `KeyValueTable--body` for vertical alignment of table itself.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.js [hunk 3]
```diff
@@ -55,54 +56,104 @@ type KeyValuesTableProps = {
...
+                // eslint-disable-next-line react/no-array-index-key
+                <tr key={`${row.key}-${i}`}>
+                  <td className="KeyValueTable--keyColumn">{row.key}</td>
+                  <td>{valueMarkup}</td>
+                  <td className="KeyValueTable--copyColumn">
+                    <Tooltip
+                      arrowPointAtCenter
+                      mouseLeaveDelay={0.5}
+                      onVisibleChange={visible => this.handleTooltipVisibilityChange(row, visible)}
+                      placement={'left'}
```
[the remark this answers is not in the document]
Comment (everett980) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.js]: I added the rule and ran `yarn eslint --fix`

**Pull request merged**